baremaps-core/src/main/java/org/apache/baremaps/storage/geoparquet/GeoParquetTypeConversion.java

```diff
@@ -69,9 +80,33 @@ public static List<Object> asRowValues(GeoParquetGroup group) {
         case DOUBLE -> values.add(group.getDoubleValue(i));
         case STRING -> values.add(group.getStringValue(i));
         case GEOMETRY -> values.add(group.getGeometryValue(i));
-        case GROUP -> values.add(null); // TODO: values.add(asDataRow(group.getGroupValue(i)));
+        case ENVELOPE -> values.add(group.getEnvelopeValue(i));
+        case GROUP -> values.add(asNested(group.getGroupValue(i)));
       }
     }
     return values;
   }
+
+  public static Map<String, Object> asNested(GeoParquetGroup group) {
+    Map<String, Object> nested = new HashMap<>();
+    Schema schema = group.getSchema();
+    List<Field> fields = schema.fields();
+    for (int i = 0; i < fields.size(); i++) {
+      Field field = fields.get(i);
+      nested.put(field.name(), switch (field.type()) {
+        case BINARY -> group.getBinaryValue(i).getBytes();
```

Comment:
When an optional field or a group is null in the geoparquet file, this will raise an exception. The same thing happens in the `asRowValues` method. I tested it with real Overture data.

Reply:
This is a good point. I'm not a fan of returning Optional everywhere, but this is something we could consider. Another option could be to return null and let the user check for it. 

@Drabble @sebr72 Which variant do you prefer?

Reply:
@bchapuis My instinct would say that when a field is optional we use Optional. This way we can make a difference with fields which are not optional with a null value (in the way our code explains it).

Reply:
@sebr72 The problem is that we also need to return Optional for each values in a potentially very big geoparquet file. As for the List, I'm a bit worried about the overhead. It also means that we need to return Optional for required values.

Here are some interesting arguments for and against the use of Optional:
https://blogs.oracle.com/javamagazine/post/optional-class-null-pointer-drawbacks

Reply:
@bchapuis based on the article and your concern about the overhead, I am happy if we don't use Optional.